### PR TITLE
Replace toWei/fromWei when base token conversion

### LINKED ISSUE #1279: Stop using toWei/fromWei from web3
We need to use amountToWei, weiToAmount to be compatible to tokens with various decimals

### PR DESCRIPTION
**Pull request by KatunaNorbert**
Fixes #1279 .

### CODE REVIEW

File: src/pools/fixedRate/FixedRateExchange.ts [hunk 1]
```diff
@@ -366,10 +366,10 @@ export class FixedRateExchange {
     const estGas = await this.estSetRate(
       address,
       exchangeId,
-      await this.unitsToAmount(exchange.baseToken, String(newRate))
+      await this.amountToUnits(exchange.baseToken, String(newRate))
```
Comment (mihaisc) [inline src/pools/fixedRate/FixedRateExchange.ts:369]: why noy toString()?

Comment (KatunaNorbert) [reply src/pools/fixedRate/FixedRateExchange.ts:369]: this is what was used above so I used to be consistent

Comment (mihaisc) [reply src/pools/fixedRate/FixedRateExchange.ts:369]: strange, we always use toString, don't know how that stayed there.

Comment (KatunaNorbert) [reply src/pools/fixedRate/FixedRateExchange.ts:369]: I will change it

Comment (bogdanfazakas) [reply src/pools/fixedRate/FixedRateExchange.ts:369]: feel free to update all String(value) in other places if you find more of this 😃

Comment (KatunaNorbert) [reply src/pools/fixedRate/FixedRateExchange.ts]: ok, I will

File: src/pools/fixedRate/FixedRateExchange.ts [hunk 2]
```diff
@@ -362,13 +362,17 @@ export class FixedRateExchange {
     exchangeId: string,
     newRate: string
   ): Promise<TransactionReceipt> {
+    const exchange = await this.getExchange(exchangeId)
     const estGas = await this.estSetRate(
       address,
       exchangeId,
-      this.web3.utils.toWei(String(newRate))
+      await this.amountToUnits(exchange.baseToken, newRate.toString())
```
Comment (bogdanfazakas) [inline src/pools/fixedRate/FixedRateExchange.ts:369]: this conversions should also take place in `estSetRate()` just pass the raw string values to the estimate method

File: src/pools/fixedRate/FixedRateExchange.ts [hunk 3]
```diff
@@ -362,13 +362,17 @@ export class FixedRateExchange {
...
     )
     const trxReceipt = await this.contract.methods
-      .setRate(exchangeId, this.web3.utils.toWei(newRate))
+      .setRate(
+        exchangeId,
+        await this.amountToUnits(exchange.baseToken, newRate.toString())
```
Comment (bogdanfazakas) [inline src/pools/fixedRate/FixedRateExchange.ts:374]: also should we have setNewRate in wei even if we have have the rate in raw string value in FreCreationParams used to create a new fixed rate for example ?

Comment (bogdanfazakas) [reply src/pools/fixedRate/FixedRateExchange.ts:374]: nevermind we convert it there also here https://github.com/oceanprotocol/ocean.js/blob/v4main/src/utils/ContractUtils.ts#L83

File: src/pools/fixedRate/FixedRateExchange.ts (hunk 2 above)
Comment (KatunaNorbert) [reply src/pools/fixedRate/FixedRateExchange.ts:369]: right, makes sense

**Comment by KatunaNorbert** [comment]
I run into an issue when using `amountToUnits` inside `createNftErc20WithFixedRate` -> `getFreCreationParams` with USDC base token to transform the fixed rate value.

When sending the rate to the contract it contains 6 zeros so `1000000` but the transaction is reverted because in the contracts we have a `MIN_RATE` check which is `10 ** 10`.
I think `amountToUnits` will not work in this case unless we change the contract.


Screenshot 2022-03-07 at 14 44 01

**Review by mihaisc — APPROVED** [review]

**Pull request merged**